cvat/apps/engine/serializers.py

```diff
@@ -1364,51 +1439,201 @@ def update(self, instance: models.Task, validated_data: dict[str, Any]) -> model
                 f"validation mode is '{models.ValidationMode.GT_POOL}'"
             )
 
-        if frame_selection_method == models.JobFrameSelectionMethod.MANUAL:
-            requested_honeypot_real_frames = validated_data['honeypot_real_frames']
+        if not frame_selection_method:
+            return instance
 
-            task_honeypot_abs_frames = (
-                instance.data.images
-                .filter(is_placeholder=True)
-                .order_by('frame')
-                .values_list('frame', flat=True)
+        # Populate the prefetch cache for required objects
+        from django.db.models import prefetch_related_objects, Prefetch
+        prefetch_related_objects([instance],
+            Prefetch('data__images', queryset=models.Image.objects.order_by('frame')),
+            'segment_set',
+            'segment_set__job_set',
+        )
+        instance.gt_job = next(
+            db_job
+            for db_segment in instance.segment_set.all()
+            for db_job in db_segment.job_set.all()
+            if db_job.type == models.JobType.GROUND_TRUTH
+        )
```

Comment:
This looks a bit odd given that there is a `gt_job` property on the task instance.
Consider refreshing cached property another way, e.g. by introducing a general method:

```python
def refresh_gt_job(self):
    field = 'gt_job'
    if field in self.__dict__:
        del self.__dict__[field]
    return self.gt_job
```